### PR TITLE
Add a basic dev container

### LINKED ISSUE #42042: Add a dev container to simplify contributing
It would be great to add a [development container](https://code.visualstudio.com/docs/remote/containers) to this repo. This can simplify getting started with the repository (since the dev container sets up all the tools you need to get started) and will also make the repo work better on codespaces

### PR DESCRIPTION
**Pull request by mjbvz**
Adds a basic dev container. This basic development environment has tools like gulp installed and should make it easier to get started contributing to TS. The dev container also will be picked for codespaces 

Fixes #42042

### CODE REVIEW

**Comment by mohameed7841987** [comment]
الغاء الاشتراك

في سبت، 19 ديسمبر، 2020 في 12:29 ص، كتب TypeScript Bot <
notifications@github.com>:

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -78,6 +78,13 @@ The TypeScript repository is relatively large. To save some time, you might want
 
 Run `gulp` to build a version of the compiler/language service that reflects changes you've made. You can then run `node <repo-root>/built/local/tsc.js` in place of `tsc` in your project. For example, to run `tsc --watch` from within the root of the repository on a file called `test.ts`, you can run `node ./built/local/tsc.js --watch test.ts`.
 
+### Using a development container
+
+This repository includes a [development container](https://code.visualstudio.com/docs/remote/containers) that you can use to quickly create an isolated development environment with all the tools you need to start working on TypeScript. To get started with a dev container and VS Code, either:
+
+- Clone the TypeScript repository locally and use the `Open Folder in Container` command.
+- Use the `Clone Repository in Container Volume` command to clone the TypeScript repository into a new container.
```
Comment (DanielRosenwasser) [inline CONTRIBUTING.md:86]: Should this one go first? Sounds like it's probably more ideal for most users, but I don't know much about containers.

Comment (mjbvz) [reply CONTRIBUTING.md:86]: Sounds good to me. Dev containers definitely make things easy once you have VS Code and docker setup, and with Codespaces you can just connect and the container will automatically be setup for you

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	// Set *default* container specific settings.json values on container create.
+	"settings": {
+		"terminal.integrated.shell.linux": "/bin/bash"
+	},
+	// Add the IDs of extensions you want installed when the container is created.
+	"extensions": [
+		"dbaeumer.vscode-eslint"
+	],
+	"remoteUser": "node"
+}
```
Comment (DanielRosenwasser) [inline .devcontainer/devcontainer.json:21]: Just for the sake of consistency, we do use spaces over tabs in the repo. Could you swap these over to spaces?

**Comment by DanielRosenwasser** [comment]
Looking at the dockerfile, I think I'd like us to use Volta instead of nvm in the future, but it's not a blocker.

**Review by kraliq233 — COMMENTED** [review]
J

**Pull request merged**